models/pardot__campaigns.sql

```diff
@@ -1,13 +1,18 @@
-{% set statuses = dbt_utils.get_column_values(table=ref('int__opportunity_tmp'), column='opportunity_status') %} 
+{% if execute and flags.WHICH in ('run', 'build') %}
+    {% set statuses = dbt_utils.get_column_values(table=ref('int__opportunity_tmp'), column='opportunity_status') %} 
+{% else %}
+    {% set statuses = [] %}
+{% endif %}
 
 with opportunities_tmp as (
     select * 
     from {{ ref('int__opportunity_tmp') }} 
     
 ), opportunities as ( 
     select 
-        source_relation, campaign_id, 
-        count(*) as count_opportunities, 
+        source_relation, campaign_id,
+        count(*) as count_opportunities
+        {% if statuses != [] %} , {% endif %} 
     
         {% for status in statuses %} 
         count(case when opportunity_status = '{{ status }}' then 1 end) as count_opportunities_{{ status|lower|replace(' ','_') }}, 
```

Comment:
Suggestion for ease of reading and maintanability: (also I couldn't edit lines 19-21, so this is meant to replace 15-21.
```suggestion
    
        {% for status in statuses if statuses != [] %} 
        , count(case when opportunity_status = '{{ status }}' then 1 end) as count_opportunities_{{ status|lower|replace(' ','_') }}
        , sum(case when opportunity_status = '{{ status }}' then amount end) as sum_opportunity_amount_{{ status|lower|replace(' ','_') }} 
        {% endfor %} 
```

Reply:
Good call, applied!